### PR TITLE
everforest-gtk-theme: init at 0-unstable-2023-03-20

### PR DESCRIPTION
**Pull request by OulipianSummer**
## Everforest GTK Theme

This pull request adds the [Everforest GTK Theme](https://github.com/Fausto-Korpsvart/Everforest-GTK-Theme/) by [Fausto-Korpsvart](https://github.com/Fausto-Korpsvart). This user has also made a number of other Vim inspired GTK themes like Gruvbox and Catppuccin which are already a part of Nixpkgs.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/data/themes/everforest-gtk-theme/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ lib, stdenv, fetchurl}:
+
+stdenv.mkDerivation rec {
+  pname = "Everforest-GTK-Theme";
```
Comment (Aleksanaa) [inline pkgs/data/themes/everforest-gtk-theme/default.nix:4]:
```suggestion
  pname = "everforest-gtk-theme";
```

File: pkgs/data/themes/everforest-gtk-theme/default.nix
Comment (Aleksanaa) [inline pkgs/data/themes/everforest-gtk-theme/default.nix:1]: This should be in `pkgs/by-name`.

File: pkgs/data/themes/everforest-gtk-theme/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  version = "1.0";
```
Comment (Aleksanaa) [inline pkgs/data/themes/everforest-gtk-theme/default.nix:5]: Where is that 1.0??? This should be `0-unstable-2024-xx-yy` with a commit date.

File: pkgs/data/themes/everforest-gtk-theme/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/Fausto-Korpsvart/Everforest-GTK-Theme/archive/refs/heads/master.tar.gz";
```
Comment (Aleksanaa) [inline pkgs/data/themes/everforest-gtk-theme/default.nix:8]: This is not pinned, and the hash can change. Point to a specific commit instead (and use `fetchFromGitHub`).

File: pkgs/data/themes/everforest-gtk-theme/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    sha256 = "2gmSMW/yUdinvZ/Jaw6GyfEL/CBqYKtw0QrYWy9KrL0=";
+  };
+
+  installPhase = ''
```
Comment (Aleksanaa) [inline pkgs/data/themes/everforest-gtk-theme/default.nix:12]: `runHook preInstall` and `runHook postInstall`.

File: pkgs/data/themes/everforest-gtk-theme/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    mkdir -p $out/share/themes
+    rm -rf README.md LICENSE icons extra
+    cp -r themes/* $out/share/themes/
+  '';
+
+  meta = with lib; {
+    description = "Everforest theme for GTK based desktop environments";
+    homepage = "https://github.com/Fausto-Korpsvart/Everforest-GTK-Theme";
+    license = licenses.gpl3Only;
```
Comment (Aleksanaa) [inline pkgs/data/themes/everforest-gtk-theme/default.nix:21]:
```suggestion
    license = licenses.gpl3Plus;
```

https://github.com/Fausto-Korpsvart/Everforest-GTK-Theme/blob/8481714cf9ed5148694f1916ceba8fe21e14937b/themes/Everforest-Dark-B/metacity-1/metacity-theme-3.xml#L6

File: pkgs/data/themes/everforest-gtk-theme/default.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{ lib, stdenv, fetchurl}:
+
+stdenv.mkDerivation rec {
```
Comment (Aleksanaa) [inline pkgs/data/themes/everforest-gtk-theme/default.nix:3]:
```suggestion
stdenvNoCC.mkDerivation rec {
```

File: pkgs/data/themes/everforest-gtk-theme/default.nix [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    mkdir -p $out/share/themes
+    rm -rf README.md LICENSE icons extra
+    cp -r themes/* $out/share/themes/
+  '';
+
+  meta = with lib; {
+    description = "Everforest theme for GTK based desktop environments";
+    homepage = "https://github.com/Fausto-Korpsvart/Everforest-GTK-Theme";
+    license = licenses.gpl3Only;
+    platforms = platforms.unix;
```
Comment (Aleksanaa) [inline pkgs/data/themes/everforest-gtk-theme/default.nix:22]:
```suggestion
    platforms = platforms.all;
```

Comment (isabelroses) [reply pkgs/data/themes/everforest-gtk-theme/default.nix:22]: Comparing to catppuccin-gtk, https://github.com/NixOS/nixpkgs/blob/980433df445ceadad8c446267dc5a9d029055239/pkgs/data/themes/catppuccin-gtk/default.nix#L89, should this not be `platforms.linux`.

Comment (Aleksanaa) [reply pkgs/data/themes/everforest-gtk-theme/default.nix:22]: That is definitely wrong, gtk themes can be used on macOS and other unix-like distributions, and even kinda work on Windows.

File: pkgs/data/themes/everforest-gtk-theme/default.nix
Comment (Leixb) [reply pkgs/data/themes/everforest-gtk-theme/default.nix:1]: It makes it so you don't need to add the package to `all-packages.nix`, AFAIK it's the new convention and should be used for new packages. See: https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md

**Pull request closed without merging**